### ISSUE TITLE
Extreme flicker with Firefox on Focal and nvidia-435

### ISSUE THREAD

**Issue by ali1234** [MirServer/mir#1309]
I am running Focal. I installed all the parts from ppa:mir-team/rc and ran miral-shell on a VT.

I started a terminal and then started Firefox.

Firefox window flickers at 60Hz: sometimes the whole window is missing in every other frame, sometimes parts of it are not redrawn properly. It is as if the compositor is re-using old (and maybe partially overwritten) frames.

**Comment by AlanGriffiths** [comment]
I've seen this with Firefox's Wayland backend. We've not yet established whether the fault is with Mir.

You can work around this by forcing Firefox to use X11. Start a terminal with `Ctrl-Alt-X` (that forces X11 for various toolkit backends) and run `firefox` from there.

**Comment by ali1234** [comment]
When I do that, the terminal renders exactly one frame and then stops. I can still type commands to launch other programs, but I do not see any text. Firefox is a full black window. Launching other terminals has the same result. Also the flicker still happens on window decorations.

**Comment by AlanGriffiths** [comment]
A pity (that does work on Intel).

We'll have to investigate further. Thanks for reporting this.

**Comment by wmww** [comment]
Looks like its time for me to get my NVIDIA kit working

**Comment by wmww** [comment]
Hmm.. the flicker seems pretty common, apparently sometimes even effecting intel GPUs. We should look into that.

Firefox on XWayland on Mir on NVIDIA on Pop!_OS 19.10 seems to work mostly fine for me. I did just now notice that XWayland has an `-eglstream` option, that I don't *think* we're using.

**Comment by AlanGriffiths** [comment]
> XWayland has an `-eglstream` option, that I don't _think_ we're using.

We're not.

```c++
    execl(
        xwayland_path.c_str(),
        xwayland_path.c_str(),
        dsp_str.c_str(),
        "-rootless",
        "-listen", abstract_socket_fd_str.c_str(),
        "-listen", socket_fd_str.c_str(),
        "-wm", wm_fd_str.c_str(),
        "-terminate",
        NULL);
```

**Comment by AlanGriffiths** [comment]
We fixed the `-eglstream` option in Mir 2.1, could someone with the problematic stack please retest?

This may also be needed:

https://discourse.ubuntu.com/t/using-firefox-wayland-on-mir-servers/19704
